src/pushsource/_impl/backend/pub_source/pub_source.py

```diff
@@ -73,26 +73,38 @@ def __exit__(self, exc_type, exc_val, exc_tb):
         self._client.shutdown()
         self._executor.shutdown(True)
 
-    def _push_items_from_ami_json(self, json_obj):
+    def _push_items_from_json(self, json_obj):
         out = []
-
-        try:
-            push_items = AmiPushItem._from_data(json_obj)
-            out.extend(list_argument(push_items))
-        except KeyError:
-            LOG.warning("Cannot parse AMI push item/s: %s", str(json_obj))
+        first_obj = next(iter(json_obj or []), None)
+        if first_obj:
+            if first_obj.get("src").endswith("vhd") or first_obj.get("src").endswith(
```

Comment:
This somewhat breaks the intended behavior of the later try/excepts of being tolerant against unknown data.

For example, if the objects had no `"src"` at all, I suppose that would have previously fallen into the "Cannot parse" warning case which is non-fatal.  But now due to this expression it would instead crash with: `'NoneType' object has no attribute 'endswith'`.

Given that there is no way to enforce the format of the files being attached to Pub tasks, that the format might change over time, and that the Pub source might eventually be used for non-AMI/VMI cases, I think it'd be best to continue being more tolerant here.  Can you find a way to write this which won't crash when "src" is not a string?